pkg/frontend/querymiddleware/codec_test.go

```diff
@@ -194,73 +198,79 @@ func TestMetricsQuery_MinMaxTime(t *testing.T) {
 		{
 			name:         "range query: without range vector, without offset",
 			metricsQuery: rangeRequest,
-			expectedMinT: startTime.UnixMilli(),
+			expectedMinT: startTime.UnixMilli() - lookbackDurationMS,
 			expectedMaxT: endTime.UnixMilli(),
 		},
 		{
 			name:         "instant query: without range vector, without offset",
 			metricsQuery: instantRequest,
-			expectedMinT: endTime.UnixMilli(),
+			expectedMinT: endTime.UnixMilli() - lookbackDurationMS,
 			expectedMaxT: endTime.UnixMilli(),
 		},
 		{
 			name:         "range query: with range vector, without offset",
 			metricsQuery: withQuery(t, rangeRequest, fmt.Sprintf("rate(go_goroutines{}[%s])", rangeVectorDurationStr)),
+			// range vector time exceeds lookback; no change from lookback duration
```

Comment:
This comment isn't quite accurate: the lookback period isn't used at all for range vector selectors.